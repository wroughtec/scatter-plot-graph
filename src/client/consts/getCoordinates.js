import gql from 'graphql-tag';

export const GET_COORDINATES = gql`
  query getCoordinates {
    coordinates {
      points
    }
  }
`;
