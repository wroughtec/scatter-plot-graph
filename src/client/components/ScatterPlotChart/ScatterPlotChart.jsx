// @flow
import React, { Component } from 'react';
import { Query } from 'react-apollo';
import Highcharts from 'highcharts';
import HighchartsReact from 'highcharts-react-official';
import { Spinner } from 'components/Spinner/Spinner';
import { GET_COORDINATES } from 'consts/getCoordinates';

export class ScatterPlotChart extends Component {
  displayChart = ({ loading, error, data }) => {
    let component = null;
    if (loading) component = <Spinner />;
    if (error) return `Error! ${error.message}`;

    if (data && data.coordinates) {
      const {
          coordinates: { points }
        } = data,
        options = {
          chart: {
            type: 'scatter',
            zoomType: 'xy'
          },
          boost: {
            useGPUTranslations: true,
            usePreAllocated: true
          },
          xAxis: {
            gridLineWidth: 1,
            title: {
              text: null
            }
          },
          yAxis: {
            minPadding: 0,
            maxPadding: 0,
            title: {
              text: null
            }
          },
          title: {
            text: 'Scatter plot of points'
          },
          legend: {
            enabled: false
          },
          series: [
            {
              name: 'Points',
              color: 'rgba(223, 83, 83, .5)',
              data: points,
              marker: {
                radius: 2
              },
              tooltip: {
                followPointer: false,
                pointFormat: 'x: {point.x:.1f}, y: {point.y:.1f}'
              }
            }
          ]
        };
      component = <HighchartsReact highcharts={Highcharts} options={options} />;
    }

    return component;
  };

  render() {
    return <Query query={GET_COORDINATES}>{this.displayChart}</Query>;
  }
}
