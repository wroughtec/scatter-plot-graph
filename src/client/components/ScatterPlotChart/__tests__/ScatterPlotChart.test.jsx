import React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import { COORDINATES_MOCK } from 'mocks/getCoordinates.mock';
import { ScatterPlotChart } from 'components/ScatterPlotChart/ScatterPlotChart';
import { render, cleanup } from 'react-testing-library';
import wait from 'waait';

describe('ScatterPlotChart', () => {
  afterEach(cleanup);

  test('should renders without error', async () => {
    const { getByText } = render(
      <MockedProvider mocks={COORDINATES_MOCK} addTypename={false}>
        <ScatterPlotChart />
      </MockedProvider>
    );

    await wait(0);

    expect(getByText('Created with Highcharts 6.2.0')).toBeTruthy();
  });

  test('should render loading state initially', async () => {
    const { container } = render(
      <MockedProvider mocks={[]}>
        <ScatterPlotChart />
      </MockedProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
