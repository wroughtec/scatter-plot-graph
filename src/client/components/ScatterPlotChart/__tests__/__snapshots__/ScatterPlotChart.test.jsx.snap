// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScatterPlotChart should render loading state initially 1`] = `
<div
  class="c-spinner"
>
  <div
    class="c-spinner__wrapper"
  >
    <svg
      class="c-spinner__icon"
      height="1em"
      preserveAspectRatio="xMinYMid meet"
      width="1em"
    >
      <use
        xlink:href="#spinner"
      />
    </svg>
  </div>
</div>
`;
